Changes related to launching renderer process via forking and stabilization

v-saha opened the pull request:
References:

1. https://github.com/Samsung/Castanets/pull/286/commits/9403f649f5c23dca3540648df278f4d6550f0d36
2. https://github.com/Samsung/Castanets/pull/286/commits/74e96280b999a4a514d59440f056126085940dcf
3. https://github.com/Samsung/Castanets/pull/286/commits/bcd321e77a20b806b59e1a0cbc1a6d530508c100
4. https://github.com/Samsung/Castanets/pull/286/commits/89adf3ec95feba596cfca2e260c477a0b282daf9
5. https://github.com/Samsung/Castanets/pull/286/commits/29620bf315fae036f784d653ec0d64154265db72
6. https://github.com/Samsung/Castanets/pull/286/commits/f6ac73c02bde763a5bcbf470f123c05d674b92b8
7. https://github.com/Samsung/Castanets/pull/286/commits/9c50f28b5191adbf3186512c30213676c8a8d252
8. https://github.com/Samsung/Castanets/pull/288/commits/1a64aa7d389c6dbadf1ac137bee9f844c6bf7c63

Commit:
Disable field trial features

Field trial testing has been enabled till now.
Some  feature such as site-per-process causes
crash. This patch disables all experimental features
for Castanets.

Signed-off-by: v-saha <v.saha@samsung.com>

Commit:
[DCHECK] AssertBlockingAllowed

base::CreateAndOpenFdForTemporaryFileInDir is called
when fd < 0.  In this scenario DCHECK is hit.
DCHECK hits when the above function is called
from a scope that disallows blocking.
To avoid this we will call the above function
using ScopedAllowIO.

Signed-off-by: v-saha <v.saha@samsung.com>

Commit:
Launch renderer process via forking

If launching via tcp connection is not established
within timeout(5 sec), use forked renderer process.

And if tcp connection is established, use that connection
and doesn't try to connect with unix domain socket.

Signed-off-by: v-saha <v.saha@samsung.com>

Commit:
Add check for tcp success callback

Fixes crash if running castanets with
--enable-forking

Signed-off-by: v-saha <v.saha@samsung.com>

Commit:
Add --tcp-launch-timeout switch

This switch specifies tcp launch timeout value to fallback
to launch renderer process via forking.

The behavior of this switch is as below.
1) out/Release/chrome --tcp-launch-timeout
Use default timeout value |kTcpLaunchTimeoutDefault|
in child_process_launcher_helper.cc

2) out/Release/chrome --tcp-launch-timeout=[timeout value]
Use the specified timeout value to fall through to launch
renderer process via unix domain socket.

Signed-off-by: v-saha <v.saha@samsung.com>

Commit:
[WorkAround] Fix the crash issue caused by OOM

New Staging buffer is created for every frame
because the QueryResult is not handled on Castanets.
It causes out of memory crash, so fall-back to glFinish.

Signed-off-by: v-saha <v.saha@samsung.com>

Commit:
[Workaround] Fix scroll performance issue

In  GPU compositing when playback is in progress
avoid waiting for query result as it delays the
rasterization process, hence for temporarily we skip it.

Signed-off-by: v-saha <v.saha@samsung.com>

suyambulingamrm reviewed (commented):
Can you take this commit also in this PR.
https://github.com/Samsung/Castanets/pull/288/commits/1a64aa7d389c6dbadf1ac137bee9f844c6bf7c63

Commit:
Fix linker error for component build

This CL has two changes as below.
1) When build with gn option 'is_component_build = true'
below error occured.

error: duplicate symbol: MojoRetryInvitation

To fix this, use name |MojoRetryInvitationImpl|
instead of |MojoRetryInvitation| at entrypoints.cc.

2) Add additional guards for CASTANETS at thunks.cc.

Signed-off-by: v-saha <v.saha@samsung.com>
